packages/pancake-uikit/src/widgets/Menu/components/UserMenu/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useRef, useState } from "react";
+import { usePopper } from "react-popper";
+import styled from "styled-components";
+import Flex from "../../../../components/Box/Flex";
+import { ChevronDownIcon } from "../../../../components/Svg";
+import isTouchDevice from "../../../../util/isTouchDevice";
+import { UserMenuProps, variants } from "./types";
+import MenuIcon from "./MenuIcon";
+import { UserMenuItem } from "./styles";
+
+const StyledUserMenu = styled(Flex)`
+  align-items: center;
+  background-color: ${({ theme }) => theme.colors.tertiary};
+  border-radius: 16px;
+  box-shadow: inset 0px -2px 0px rgba(0, 0, 0, 0.1);
+  cursor: pointer;
+  display: inline-flex;
+  height: 32px;
+  padding-left: 40px;
+  padding-right: 8px;
+  position: relative;
+
+  &:hover {
+    opacity: 0.65;
+  }
+`;
+
+const LabelText = styled.div`
+  color: ${({ theme }) => theme.colors.text};
+  display: none;
+  font-weight: 600;
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    display: block;
+    margin-left: 8px;
+    margin-right: 4px;
+  }
+`;
+
+const Menu = styled.div`
+  background-color: ${({ theme }) => theme.card.background};
+  border: 1px solid ${({ theme }) => theme.colors.cardBorder};
+  border-radius: 16px;
+  padding-bottom: 4px;
+  padding-top: 4px;
+  width: 280px;
+  z-index: 101;
+
+  ${UserMenuItem}:first-child {
+    border-radius: 8px 8px 0 0;
+  }
+
+  ${UserMenuItem}:last-child {
+    border-radius: 0 0 8px 8px;
+  }
+`;
+
+const UserMenu: React.FC<UserMenuProps> = ({
+  account,
+  text,
+  avatarSrc,
+  variant = variants.DEFAULT,
+  children,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [targetRef, setTargetRef] = useState<HTMLDivElement | null>(null);
```

Comment:
I think `useState<HTMLDivElement>()` works as well

Reply:
Works on the FE because we specifically have `strictNullChecks` set to false. The toolkit has `strict` as true so it makes `strictNullChecks` true.